tests/multinode.rs

```diff
@@ -389,28 +390,59 @@ fn test_multi_node_dynamic_network() {
         send_tx_and_retry_get_balance(&leader_data, &alice, &bob_pubkey, Some(1000)).unwrap();
     assert_eq!(leader_balance, 1000);
 
-    let validators: Vec<(NodeInfo, FullNode)> = (0..N)
+    let validators_per_thread = Arc::new(RwLock::new(Vec::new()));
+    let threads: Vec<_> = (0..10)
         .into_iter()
-        .map(|n| {
-            let keypair = KeyPair::new();
-            let validator = TestNode::new_localhost_with_pubkey(keypair.pubkey());
-            let rd = validator.data.clone();
-            //send some tokens to the new validator
-            let bal =
-                send_tx_and_retry_get_balance(&leader_data, &alice, &keypair.pubkey(), Some(500));
-            assert_eq!(bal, Some(500));
-            let val = FullNode::new(
-                validator,
-                false,
-                LedgerFile::Path(ledger_path.clone()),
-                Some(keypair),
-                Some(leader_data.contact_info.ncp),
-            );
-            info!("started[{}/{}] {:x}", n, N, rd.debug_id());
-            (rd, val)
```

Comment:
Checkout into_par_iter 
https://github.com/solana-labs/solana/blob/master/src/bin/client-demo.rs#L131

It’s part rayon

Reply:
Thanks @aeyakovenko , that made it really simple.